google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -829,6 +861,22 @@ public Builder setEndpoint(String endpoint) {
       return this;
     }
 
+    /** Gives the ability to enable gRPC compression. */
+    public Builder setEnableCompression(boolean enableCompression) {
+      this.enableCompression = enableCompression;
+      return this;
+    }
+
+    /**
+     * Gives the ability to set the threshold in bytes above which gRPC compression happens. But to
+     * enable gRPC compression at the first place, setEnableCompression(true) should be called along
+     * with this method.
+     */
+    public Builder setCompressionBytesThreshold(long compressionBytesThreshold) {
+      this.compressionBytesThreshold = compressionBytesThreshold;
```

Comment:
To enforce the `setEnableCompression` you could use the `Preconditions` class

Reply:
very valid point. I have added a Precondition check and also a unit test. Thanks!

Reply:
you are correct that the state won't change after instantiation, however, I don't think Precondition checks can be added in a builder as there is no order in calling its set methods.